docling_core/types/doc/document.py

```diff
@@ -2348,6 +2701,71 @@ def iterate_items(
                     included_content_layers=my_layers,
                 )
 
+    def iterate_items_with_stack(
```

Comment:
- first let's double check if this is really needed (see PR comment on exposing "`stack`" on the API)
- if needed, let's check:
    - if it should be part of the public API
    - what we can reuse with `iterate_items()`

Reply:
I think that the original iterate_items can just reuse the `iterate_items_with_stack` where the level = len(stack)

Reply:
I fixed the duplicate code.